templates/tfengine/components/iam_members/variables.tf

```diff
@@ -0,0 +1,49 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+variable "storage_bucket_iam_members" {
+  description = "IAM members for storage buckets. Assigns IAM bindings to a list of storage bucket names."
+  default     = {}
+  type        = map(object({
```

Comment:
Can you confirm why these are not the same?

```
type = map(object(...
```

```
type = object({...
```

Apply for all the others

Reply:
They are not the same. For example:
```
type = object(
      field = string
)
```
can be applied to `{field = "foo"}`
while
```
type = map(object(
      field = string
))
```
can be applied to `{"bar" = {field = "foo"}}`

Reply:
Got it, my bad.

I'm resolving